(docs) Clarify Hiera data loading docs (#2200)
puppetlabs/bolt: documentation/applying_manifest_blocks.md

> Following the Hiera 5 convention, the default data dir is relative to

Reviewer: ```suggestion
Following the Hiera 5 convention, the default data directory is relative to
```

> `hiera.yaml` at `$BOLTDIR/data`. For configfile examples, see [Configuring

Reviewer: ```suggestion
`hiera.yaml` at `$BOLTDIR/data`. For configuration file examples, see [Configuring
```

Reviewer: When we use `$BOLTDIR` here, are we talking specifically about an embedded project directory, or is $BOLTDIR an environment variable? If we're just talking about a Bolt project directory, then maybe we can just use <PROJECT_DIRECTORY>?

Author: No, I think it's what we used to use for `<project_directory>`. Should it be all caps?

Author: It looks like elsewhere we use PROJECT_NAME https://puppet.com/docs/bolt/latest/bolt_running_plans.html#how-bolt-locates-plans

Author: And <project-directory> https://puppet.com/docs/bolt/latest/configuring_bolt.html#bolt-projectyaml

Reviewer: Good point. Let's standardize it as `<PROJECT_DIRECTORY>` so we're following Puppet style. We should change `<PROJECT_NAME>` anyway, because technically your project directory name and project name don't have to match.

Author: Ended up with `<PROJECT DIRECTORY>` which seems clearest and inline with style guidelines.

> Hiera](https://puppet.com/docs/puppet/latest/hiera_config_yaml_5.html). Bolt 
> will not automatically load any Hiera data from this directory, you must
> set paths that you want loaded in your Hiera config.

Reviewer: ```suggestion
Hiera](https://puppet.com/docs/puppet/latest/hiera_config_yaml_5.html). Bolt 
does not automatically load Hiera data from this directory. If you want want to 
load data from a Hiera directory, you must set the relevant paths in your Hiera 
configuration file.
```

Reviewer 2: There is 2 times want on line 250

Reviewer: Ah thanks!
